### DISCUSSION TITLE
How do you write cross-platform Bleak code, that uses the OS-specific scanners and options?

### DISCUSSION THREAD

**Discussion by victorhooi** [hbldh/bleak#1324] [Q&A]
Hi,

I'm trying to use the [pytimeflip](https://github.com/pierre-24/pytimefliplib) library to interface with my [Timeflip](https://timeflip.io/) device.

My computer is running macOS.

However, it seems that macOS does BLE a bit different - the first issue I hit is that it seems macOS doesn't return normal MAC addresses by default (https://github.com/hbldh/bleak/issues/284, https://github.com/hbldh/bleak/issues/140).

However, I saw https://github.com/hbldh/bleak/pull/1073, and it seems like I can use the [macOS specific Bleak scanner](https://bleak.readthedocs.io/en/latest/backends/macos.html
), which could potentially get me MAC addresses (and I assume would work better under macOS).

My question is - what is the recommended (or best-practice) way to write Bleak code, that handles the different platform backends (e.g. Windows vs Linux vs macOS), and with specific arguments (e.g. `use_bdaddr`)?

**Comment by dlech** [comment]
Bleak automatically uses the correct backend for your OS. If this is for a personal project, then sure, use `use_bdarrd` if you want to, but don't count on it working for everyone since it is a hack. It would be best to just use the UUID returned as the ID on macOS instead.
